components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/UnsupportedAddonsAdapter.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.feature.addons.Addon
+import mozilla.components.feature.addons.AddonManager
+import mozilla.components.feature.addons.R
+
+/**
+ * An adapter for displaying unsupported add-on items.
+ *
+ * @property addonManager Manager of installed and recommended [Addon]s and manages their states.
+ * @property unsupportedAddonsAdapterDelegate Delegate that will provides method for handling the
+ * callbacks of uninstalling an add-on.
+ * @property addons The list of unsupport add-ons based on the AMO store.
+ */
+class UnsupportedAddonsAdapter(
+    private val addonManager: AddonManager,
+    private val unsupportedAddonsAdapterDelegate: UnsupportedAddonsAdapterDelegate,
+    private val addons: List<Addon>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        holder as UnsupportedAddonViewHolder
+        val addon = addons[position]
+
+        holder.titleView.text =
+            if (addon.translatableName.isNotEmpty()) {
+                addon.translatableName.translate()
+            } else {
+                addon.id
+            }
+
+        holder.removeButton.setOnClickListener {
```

Comment:
I think we should add an additional function to the delegate as deleting is a terminal operation (The unsupported add-ons and all its related data come ones after migration, if the user delete on by mistake it will be lost for ever) here we may want to give the app an opportunity to show a confirmation dialog, the app could tell us if the user want to delete permanently. This is not a blocker if we think it will take a lot of time we can address it in follow-up

Reply:
Dropping this since we can address as a follow up in the future.

Reply:
👍